### PR TITLE
Decouple calculate fee from Wallet

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/2352

### CODE REVIEW

File: src/neo/Wallets/Wallet.cs [hunk 1]
```diff
@@ -347,79 +346,12 @@ private Transaction MakeTransaction(DataCache snapshot, byte[] script, Signer[]
                     tx.SystemFee = engine.GasConsumed;
                 }
 
-                tx.NetworkFee = CalculateNetworkFee(snapshot, tx);
+                tx.NetworkFee = CalculateNetworkFee(snapshot, tx, ProtocolSettings, (a) => GetAccount(a)?.Contract?.Script);
```
Comment (Qiao-Jin) [inline src/neo/Wallets/Wallet.cs:349]: It seems we still need to rely on `Wallet` for func CalculateNetworkFee

Comment (shargon) [reply src/neo/Wallets/Wallet.cs:349]: You can use this method if you have a dictionary

File: src/neo/Wallets/Helper.cs [hunk 2]
```diff
@@ -39,5 +44,72 @@ internal static byte[] XOR(byte[] x, byte[] y)
...
+        {
+            UInt160[] hashes = tx.GetScriptHashesForVerifying(snapshot);
+
+            // base size for transaction: includes const_header + signers + attributes + script + hashes
+            int size = Transaction.HeaderSize + tx.Signers.GetVarSize() + tx.Attributes.GetVarSize() + tx.Script.GetVarSize() + IO.Helper.GetVarSize(hashes.Length);
+            uint exec_fee_factor = NativeContract.Policy.GetExecFeeFactor(snapshot);
+            long networkFee = 0;
+            foreach (UInt160 hash in hashes)
+            {
+                byte[] witnessScript = accountScript(hash);
```
Comment (vncoelho) [inline src/neo/Wallets/Helper.cs:58]: Looks good, thus we are passing the function that has the local storages.

**Review by vncoelho — APPROVED** [review]
Looks good to me.
It opens more possibilities, in my opnion.

**Comment by vncoelho** [comment]
@shargon, my idea was that with this change we would be able to, in the future, to add  a new call on neo-modules RPCServer to make this calculation.

However, we need to discuss if this is a good practice.
One thing is to create the `script` of a transaction with `InvokeFunction`.
Another thing is to really pass `verificationScripts` beforehand.
In my opinion, there is not much problem because, anyway, seed nodes will have it in advance of consensus.

**Comment by superboyiii** [comment]
@erikzhang  Merge?

**Comment by cschuchardt88** [comment]
@shargon Wasn't this fixed already?

**Comment by Jim8y** [comment]
Close as inactive

**Comment by shargon** [comment]
I want to do this, it  doesn't hurt I think

**Comment by shargon** [comment]
Moved to https://github.com/neo-project/neo/pull/3147

**Pull request closed without merging**